web/app/js/components/TopModule.jsx

```diff
@@ -16,7 +16,8 @@ class TopModule extends React.Component {
       resource: PropTypes.string.isRequired
     }).isRequired,
     startTap: PropTypes.bool.isRequired,
-    updateNeighbors: PropTypes.func
+    updateNeighbors: PropTypes.func,
+    updateTapClosingState: PropTypes.func.isRequired
```

Comment:
I think this prop should be not required, with a default value of `_.noop`. The resource detail pages use the Top module, but don't use a form, so we don't need to update the form stop button in this case.

When switching from a ResourceDetail page to another page, we get a console error because that page doesn't set the updateTapClosingState function
```
Warning: Failed prop type: The prop `updateTapClosingState` is marked as required in `TopModule`, but its value is `undefined`.
    in TopModule (at AppContext.jsx:10)
    in Unknown (at ResourceDetail.jsx:218)
    in div (at ResourceDetail.jsx:217)
    in div (at ResourceDetail.jsx:197)
    in div (at ResourceDetail.jsx:265)
    in div (at ResourceDetail.jsx:264)
    in ResourceDetailBase (at AppContext.jsx:10)
    in Unknown (created by Route)
    in Route (at index.js:53)
    in Switch (at index.js:47)
    in div (at index.js:46)
    in div (created by Basic)
    in Basic (created by Adapter)
```

Reply:
Ah, good catch!